### PR TITLE
Vendor gardener v1.59.0 and enable heartbeat controller

### PR DESCRIPTION
**Pull request by AleksandarSavchev**
**How to categorize this PR?**

/area networking
/kind enhancement

**What this PR does / why we need it**:
Vendor gardener v1.59.0.
This PR enables new heartbeat controller and removes unnecessary replace directives from go.mod file.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/4798

**Special notes for your reviewer**:

**Release note**:

```other dependency
Dependency `github.com/gardener/gardener` is updated `v1.54.0` -> `v1.59.0`.
```
```other operator
Adds a heartbeat controller that creates and renews a `Lease` resource named `gardener-extension-heartbeat` in the namespace where the extension controller is deployed. This `Lease` resource can be used by other controllers to check if the `networking-cilium` extension controller is running. 
```

### CODE REVIEW

**Review by plkokanov — APPROVED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -111,46 +111,32 @@ require (
...
-	k8s.io/apiserver => k8s.io/apiserver v0.24.4
-	k8s.io/autoscaler => k8s.io/autoscaler v0.0.0-20220531185024-cc90d57b7fe1 // translates to k8s.io/autoscaler/vertical-pod-autoscaler@v0.11.0
-	k8s.io/autoscaler/vertical-pod-autoscaler => k8s.io/autoscaler/vertical-pod-autoscaler v0.11.0
-	k8s.io/client-go => k8s.io/client-go v0.24.4
-	k8s.io/code-generator => k8s.io/code-generator v0.24.4
-	k8s.io/component-base => k8s.io/component-base v0.24.4
-	k8s.io/helm => k8s.io/helm v2.16.1+incompatible
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.24.4
-	sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.12.1
-)
```
Comment (shafeeqes) [inline go.mod:156]: Shouldn't we still need this, to keep the values in sync with gardener, in case some other module is importing it?

**Review by ScheererJ — APPROVED** [review]
/lgtm

File: go.mod (hunk 1 above)
Comment (shafeeqes) [reply go.mod:156]: Sure, makes sense. Thanks for explaining.

**Review by DockToFuture — APPROVED** [review]
/lgtm

**Pull request merged**